QCvault

JeromeMartinez opened the pull request:
Options for a cache directory instead of default sidecar file.
Cache directory may also be accessed by 2 different computers having 2 different root paths for the content directory.
Cache directory may be set per qcli call (option at each call) or for all qcli calls (configure once then all other calls use it).

CLI:
```
-cd <cache directory path>
    Use the indicated path as the cache directory location.
    Use "-cd default" for using the standard cache directory location.
-set-cd <cache directory path>
    Register the indicated path as the cache directory location.
    Use "-set-cd default" for using the standard cache directory location.
-show-cd
    Show the registered cache directory location.
-clear-cd
    Clear the cache directory location. Default directory becomes the input directory.
```

GUI:
![image](https://user-images.githubusercontent.com/1641638/96748346-b3c02380-13c9-11eb-992e-1c202866dbea.png)

dericed commented:
The default system cache directory is a little buried and hard to get to. I suggest a button in preferences directory tab or a toolbar menu option to open the cache directory in the system file browser.

dericed commented:
On my system, the default cache dir is ~/Library/Application Support/MediaArea/QCTools/cache. This folder doesn't exist and the process of trying to save to the default cache directory doesn't seem to create it or provide an error message.

dericed commented on Source/Cli/cli.cpp:
```diff
@@ -109,6 +203,16 @@ int Cli::exec(QCoreApplication &a)
                 << "-a" << std::endl
                 << "    All (stats + thumbnails + panels)." << std::endl
                 << "    Is default." << std::endl
+                << "-cd <cache directory path>" << std::endl
```
I'm wondering if `cache directory` is a good name or if `report directory` would be more clear.

dericed replied:
@bturkus likes QCvault

dericed commented on Source/Cli/cli.cpp:
```diff
@@ -204,6 +306,30 @@ int Cli::exec(QCoreApplication &a)
+        {
+            if (!output.isEmpty())
+            {
+                std::cout << "-cd and -o can not be used at same time, analyzing aborted." << std::endl;
```
rather use the term 'stopped'.

dericed commented on Source/GUI/preferences.cpp:
```diff
@@ -116,6 +126,13 @@ void PreferencesDialog::Load()
+    else if (cacheDirectoryPathString()==defaultCacheDirectoryPathString())
+        ui->CacheDir_Default->setChecked(true);
+    else
+        ui->CacheDir_Custom->setChecked(true);
```
Can the custom directory be relative to the source file. Like the the source is /a/b1/c.mkv and custom is ../b2, then would output be

```
/a/b1/c.mkv
/a/b2/c.qctools.mkv
```

JeromeMartinez commented:
> The default system cache directory is a little buried and hard to get to. I suggest a button in preferences directory tab or a toolbar menu option to open the cache directory in the system file browser.

Done.

> On my system, the default cache dir is ~/Library/Application Support/MediaArea/QCTools/cache. This folder doesn't exist and the process of trying to save to the default cache directory doesn't seem to create it or provide an error message.

Fixed.

> @bturkus likes QCvault

Changed.

> rather use the term 'stopped'.

Definitely better, changed.

> Can the custom directory be relative to the source file. Like the the source is /a/b1/c.mkv and custom is ../b2, then would output be

For the moment, only absolute paths are well supported, relative paths would be in a next PR, IMO not preventing a merge.

***

@dericed also rebased, please revieww again.

dericed commented:
I tested with:

```
% ./qctools-cli/qcli -set-qcvault hi
% ./qctools-cli/qcli -show-qcvault
QCvault path is hi.

% ./qctools-cli/qcli -i test.mkv
qcli dev build, Oct 27 2020 00:25:09
filters selected: signalstats cropdetect astats 

analyzing input file... test.mkv
.................................................. 100 of 100 %
analyzing completed

generating QCTools report... 
.................................................. 100 of 100 %
adding thumbnails to QCTools report... 
.................................................. 100 of 100 %
adding panels to QCTools report... 
.................................................. 100 of 100 %
generating QCTools report... done, in hi/test.mkv.build.QtCreator.Project.qctools.GitHub.Documents.davidrice.Users.qctools.mkv
```

The output filename is unexpected, `hi/test.mkv.build.QtCreator.Project.qctools.GitHub.Documents.davidrice.Users.qctools.mkv`
